### ISSUE TITLE
Can you make a game speed adjustable?

### ISSUE THREAD

**Issue by ChekPuk** [endless-sky/endless-sky#4818]
Hello.

Can you make game speed adjustable? I want it to be slooow.

I've beeing playing this lovely game on old laptop for some time. Laptop is very old, OpenGL 3.1 is "simulated" with llvmpipe and all actions in game are sloooow. But it's so realistic. Turning ships taking time. They are slow, you feel weight, you feel the need of upgrade, because they are so slow. I get just to it, I like it that way! And recently my son wanted to play it to, so we installed it on his new and shiny laptop and... it is blazing fast. Ships are jumping like in flee circus on screen, shooting like a hell, no feel of mass, no atmosphere of far distances at all. It runs about 4 times faster then I get just to, I didn't even knew that this game is that fast :)

Please, can you make game speed adjustable?

**Comment by Zitchas** [comment]
As someone who plays on a fairly fast machine, I second this. Battles are often over too fast, and everything feels like a fighter or a fast attack destroyer. Being able to slow the game down (without sacrificing the ability to double speed) would be nice.

**Comment by tehhowch** [comment]
Please verify that the "fast forward" setting is not toggled on (by default, this is bound to Caps Lock but can be adjusted in self-compiled builds as of the last few weeks). When on, the following is displayed in the top left: 
![image](https://user-images.githubusercontent.com/20871346/74598191-8baa4480-5032-11ea-9111-cd6b35bb5023.png)

With this on, the game runs at 3x speed.

**Comment by ChekPuk** [comment]
I have latest release 0.9.10 I see no "fast forward" option. Tried on both, on Linux and on Windows - same speed, so I think it is at normal speed now. It;s just to fast on my opinion.

So you say you made an option to increase speed even more? 3 times? O_o
Man! Slow it down! It will make a hole in my monitor :)

**Comment by ChekPuk** [comment]
Tried github version, If I press "fast forward" it goes even faster indeed. To much. With this kind of gameplay you can just turn off flying at all, switch from port to port and trade. Slower mode is more time consuming, but it's a lot more enjoyable. Is there alternative with such gameplay?

**Comment by MCOfficer** [comment]
> Tried github version, If I press "fast forward" it goes even faster indeed. To much. With this kind of gameplay you can just turn off flying at all, switch from port to port and trade. Slower mode is more time consuming, but it's a lot more enjoyable. Is there alternative with such gameplay?

It's not exactly an intended feature, but you can run Endless Sky from the command line with the `-d` switch:
```
endless-sky -d
```

This will enable debug mode, which inverts the functionality of the capslock key - it will be 3 times *slower* than usual.

**Comment by ChekPuk** [comment]
> ```
> endless-sky -d
> ```
> 
> This will enable debug mode, which inverts the functionality of the capslock key - it will be 3 times _slower_ than usual.

Capslock switches keyboard layout in my setup, so I've set "fast forward" to other key. Then started 'endless-sky -d' as you described, and trick didn't work for me. It is still gets faster when I turn "fast forward". Tomorrow I'll try again and post console output.

**Comment by tehhowch** [comment]
Debug mode slowdown is always bound Caps Lock state. It is not a general-purpose use case to launch the game in debug mode, and no configuration of the debugging slowdown key is planned.

**Comment by Zitchas** [comment]
That's unfortunate. It would be really nice to be able to have an alternate key for slowing things down.

For general travel, I use 2x speed all the time. Even with fast ships, I tend to spend most of my time in super-speed mode.

For combat, I find everything goes far too fast, and I find myself wishing repeatedly that I had a "slow down" button. 

My ideal scenario would be to have caps-locks attached to a pull down menu in the preferences that sets it to either 50% or 200% speed; and then have a user-selectable button that sets the game to the other extreme. (So, caps-locks could be used to reduce the speed by half, and P doubles the speed.)

**Comment by Amazinite** [comment]
I can imagine having the “fast forward” key be changed to a “change game speed” key that when pressed cycles through 0.5x (or 0.33x, whichever debug is), 1x, and 3x speed. Or perhaps “game speed” could even be a preference. The game already has various mechanics one would find in RTS games that it’s implemented over the years: drag-select escorts, put escorts into groups bound to number keys, zoom in and out to get a better view of the “battlefield.” Game speed is an option I remember from the old Command and Conquer games. 
I feel that a slower game speed had been requested enough that it’s worth exploring variable game speed as a mechanic.

**Comment by Zitchas** [comment]
I hadn't thought of that, but yes, having a single key rotate through a range of speeds would work very nicely. That being said, I would have it go down. So game starts at 1x, goes down to 0.5x, then loops back to the top at 3x, down to 2x, to 1x, etc. 

The reason is that, generally, needing to slow things down is a sudden thing, whereas needing to speed things up is a gradual thing. Thus making things slower should be easier (and faster) to do than speeding things up.

**Comment by int-ua** [comment]
Reasons to have slower speed option are explained in great detail in these videos by Extra Credits (a YouTube channel about game design and more):
[Accessibility - The Curb Cut Effect](https://www.youtube.com/watch?v=PJoax1Z1x4Y)
[Depth vs Complexity - Why More Features Don't Make a Better Game at 3:27](https://youtu.be/jVL4st0blGU?t=207)

In short: I'm mostly playing this game for story. A slowdown or pause even without ability to change things would've been beneficial. At current state the space UX is mostly frustrating at the start, especially taking into account that only a small part of interface is explained and you are left wondering what do the meters mean as time whizzes by causing anxiety and irritation.

~~Also, looks like AppImage does not recognize the `-d` key. Is it supposed to work with AppImage?~~ I thought it's not hardcoded to be CapsLock, will try again later.

Why not use + and - for speed control as they are in a lot of games?

**Comment by Amazinite** [comment]
+ and - are right now used for zooming with the keyboard.

**Comment by nkiest** [comment]
I agree. New Endless Sky player, old EV/EVO/EVN player, combat just flies by faster than I can register on my nice gaming rig. I'm not sure if it is a game speed thing exactly, or a general DPS to defense ratio, however. AI just blasts everything in seconds, no chance for clever maneuvers.

**Comment by jchamlin** [comment]
You can use your GPU's preferences to limit the frame rate of the game to 45FPS or 30FPS or something, and that will have the effect of slowing the game down 25% or 50% if you want to have a more manageable experience. You can then use fast forward to speed it up for the boring parts. That's what I do. I'll look into what it would take to make the game speed adjustable and decoupled from the framerate.

**Comment by tux2603** [comment]
On it. See [my fork's main project](https://github.com/tux2603/endless-sky/projects) for progress
